log10/anthropic.py

```diff
@@ -1,39 +1,50 @@
 import os
 from copy import deepcopy
+import time
 from typing import List
-from log10.llm import LLM, ChatCompletion, Message, TextCompletion
+from log10.llm import LLM, ChatCompletion, Kind, Message, TextCompletion, merge_hparams
 
 
 from anthropic import HUMAN_PROMPT, AI_PROMPT
 import anthropic
 
 import uuid
 
-import logging
-
 
 class Anthropic(LLM):
-    def __init__(self, hparams: dict = None, skip_initialization: bool = False):
+    def __init__(
+        self, hparams: dict = None, skip_initialization: bool = False, log10_config=None
+    ):
+        super().__init__(hparams, log10_config)
+
         if not skip_initialization:
             self.client = anthropic.Anthropic()
-        self.hparams = hparams
 
         if "max_tokens_to_sample" not in self.hparams:
             self.hparams["max_tokens_to_sample"] = 1024
 
     def chat(self, messages: List[Message], hparams: dict = None) -> ChatCompletion:
-        completion = self.client.completions.create(
-            **self.chat_request(messages, hparams)
+        request = self.chat_request(messages, hparams)
+
+        # Log a request which conforms wiht OpenAI.
+        merged_hparams = merge_hparams(hparams, self.hparams)
+        completion_id = self.log_start(
+            {"messages": [message.to_dict() for message in messages], **merged_hparams},
+            Kind.chat,
         )
+
+        # Carry out completion
```

Comment:
Consider pulling out the timed completion pattern into a separate function call

Reply:
Will go away if we remove the openai/anthropic traced "abstractions". Re: removing the `log10(anthropic)` and `log10(openai)` from the examples.


We are using the abstraction in the camel agent anyway, so may not hurt to land. But the question is whether we port camel to use langchain llm abstraction as well.